### ISSUE TITLE
Issue with unary constructors with `char` member.

### ISSUE THREAD

**Issue by jimbxb** [pschachte/wybe#147]
Unary constructors with `char` (or other < word size types) do not behave correctly in the presence of other constructors.

e.g.
```
constructors ctor(c:char) | remove_me(t:_) | const
def check(x:_):string = if { x = ctor(_) :: "correct" | otherwise :: "incorrect" }
!println('1'^ctor^check)
```

The expected output is `correct` but instead is `incorrect`.

Removing `remove_me` corrects the behaviour.

**Comment by jimbxb** [comment]
More oddities:

```
constructors ctor(b:bool) | remove_me(t:_) | const
def check(x:_):string = if { x = ctor(_) :: "correct" | otherwise :: "incorrect" }
!println(check(ctor(false)))
```
Compiling results in:
```
EncodeException "The serialized GlobalReference Name \"test.check<0>\" has type PointerType {pointerReferent = FunctionType {resultType = IntegerType {typeBits = 64}, argumentTypes = [IntegerType {typeBits = 1}], isVarArg = False}, pointerAddrSpace = AddrSpace 0} but should have type PointerType {pointerReferent = FunctionType {resultType = IntegerType {typeBits = 64}, argumentTypes = [IntegerType {typeBits = 64}], isVarArg = False}, pointerAddrSpace = AddrSpace 0}"
```

**Comment by jimbxb** [comment]
Interesting finds:

Using the original code (with `char`), I found that every fourth ASCII ('d', 'h') character prints "correct". 

In the implicit equality test, the tag is the lowest bit, and the character is shifted left by one, ored with 512 (not important) 
so the representation of `ctor('a')` is `10110000100`, and `ctor('d')` is `1011001000`. 
The value is anded with 7 (111), which is from `tagMask` in `Config.hs`, and will succeed if it is 0, explaining why every 4th succeeds.

Changing the value of `tagMask` to fixes the issue, however I don't know if this is actually a fix.

**Comment by pschachte** [comment]
Fixed in commit #151
